src/domain/kaivuilmoitus/components/ApplicationReportOperationalConditionDialog.tsx

```diff
@@ -0,0 +1,192 @@
+  isOpen: boolean;
+  onClose: () => void;
+  applicationId: number;
+};
+
+const ApplicationReportOperationalConditionDialog: React.FC<Props> = ({
+  isOpen,
+  onClose,
+  applicationId,
+}) => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { data: application, refetch } = useApplication(applicationId);
+  const formContext = useForm<ReportOperationalConditionData>({
+    mode: 'onTouched',
+    resolver: yupResolver(reportOperationalConditionSchema),
+    context: {
+      application: application,
+      dateBeforeStartErrorMessageKey: 'dateBeforeStart',
+      dateInFutureErrorMessageKey: 'dateInFuture',
+    },
+  });
+  const { handleSubmit, reset: resetForm, formState } = formContext;
+  const isConsifmButtonEnabled = formState.isValid;
+  const {
+    mutate,
+    reset: resetMutation,
+    isLoading,
+    isError,
+  } = useMutation(reportOperationalCondition);
+  const { showReportOperationalConditionSuccess, showReportOperationalConditionError } =
+    useApplicationReportOperationalConditionNotification();
+  const dialogTitle = t('hakemus:operationalConditionDialog:title');
+  const { id, applicationData, ilmoitukset } = application as Application;
+  const previousReports = ilmoitukset?.TOIMINNALLINEN_KUNTO ?? ([] as Ilmoitus[]);
+  const startDate = applicationData.startTime as Date;
+  const today = new Date();
+
+  useEffect(() => {
+    if (isOpen) {
+      refetch().then(() => {
+        formContext.setValue('applicationId', applicationId);
+      });
+    }
+  }, [isOpen, refetch, formContext, applicationId]);
+
+  function handleClose() {
+    resetForm();
+    resetMutation();
+    onClose();
+  }
+
+  async function submitForm(data: ReportOperationalConditionData) {
+    mutate(data, {
+      onSuccess() {
+        showReportOperationalConditionSuccess();
+        handleClose();
+      },
+      onError() {
+        showReportOperationalConditionError();
+        handleClose();
+      },
+    });
+  }
+
+  return (
+    <Dialog
+      id="application-report-operational-condition"
+      isOpen={isOpen}
+      aria-labelledby={dialogTitle}
+      variant="primary"
+      close={onClose}
+      closeButtonLabelText={t('common:ariaLabels:closeButtonLabelText')}
+    >
+      <Dialog.Header
+        id="application-report-operational-condition-title"
+        title={dialogTitle}
+        iconLeft={<IconInfoCircleFill aria-hidden="true" />}
+      />
+      <FormProvider {...formContext}>
+        <form onSubmit={handleSubmit(submitForm)}>
+          <Dialog.Content>
+            <Instructions previousReports={previousReports} />
+            <VisuallyHiddenInput
+              name="applicationId"
+              value={id as number}
+              aria-hidden="true"
+              readOnly
+            />
+            <DatePicker
+              name="date"
+              label={t(`hakemus:operationalConditionDialog:date`)}
+              locale={locale}
+              minDate={startDate}
```

Comment:
minDate restriction does not seem to work here. I am able to select not allowed dates in date picker.

Reply:
What might cause this and do you have any idea how to fix it? Typing the date works correctly.

Reply:
You're right. I have trouble sometimes to spot this kinds of errors. I cannot see how it can be a string if the original field is of type Date. But I fixed it now by explicitely defining startDate as a Date: ` const startDate = new Date(applicationData.startTime?.toString() ?? 0);`

Reply:
I guess the type of the field might be wrong, but there was some reason why it's defined as Date which I can't remember now.